test/unit/location.d

```diff
@@ -0,0 +1,35 @@
+module location;
+
+import dmd.lexer;
+import dmd.globals : Loc;
+
+unittest
+{
+    string text = "int a b xyz\nfloat\n\nchar a=b + c[11];";
+    scope Lexer lex1 = new Lexer(null, text.ptr, 0, text.length, 0, 0);
+    auto expected = [
+        [1, 1, 0],
+        [1, 5, 4],
+        [1, 7, 6],
+        [1, 9, 8],
+        [2, 1, 12],
+        [4, 1, 19],
+        [4, 6, 24],
+        [4, 7, 25],
+        [4, 8, 26],
+        [4, 10, 28],
+        [4, 12, 30],
+        [4, 13, 31],
+        [4, 14, 32],
+        [4, 16, 34],
+        [4, 17, 35],
+    ];
+
+    foreach(exp; expected)
+    {
+        lex1.nextToken();
+        assert(lex1.token.loc.linnum == exp[0]);
+        assert(lex1.token.loc.charnum == exp[1]);
+        assert(lex1.token.loc.offset == exp[2]);
```

Comment:
I know some will not agree with me, but in my opinion, it's best practice to have one assertion per unit test block. That means one unit test block per test case. That would be one unit test block per token, plus a few more for the edge cases. Again, have a look at my implementation: https://github.com/jacob-carlborg/ddc/commit/32d5fe37be211065558f6b44733f059ec9f10c13#diff-ca42ad79bd703d8bea0e34b342881ac9.